hack/dev-setup

```diff
@@ -0,0 +1,122 @@
+#!/bin/sh
+
+DEV_DIR=$(dirname "${0}")/../dev
+EXAMPLE_DIR=$(dirname "${0}")/../example
+MINIKUBE_IP=$(minikube ip)
+IP_ROUTE=$(ip route get 1)
+IP_ADDRESS=$(echo ${IP_ROUTE#*src} | awk '{print $1}')
+
+echo "After OSX sleep, minikube's time is off."
+echo "This results in x509 certificate auth to not work for shoot certificates"
+minikube ssh -- docker run -i --rm --privileged --pid=host debian nsenter -t 1 -m -u -n -i date -u $(date -u +%m%d%H%M%Y)
+
+mkdir -p ${DEV_DIR}
+
+cp ${EXAMPLE_DIR}/componentconfig-gardener-controller-manager.yaml ${DEV_DIR}/
+
+kubectl apply -f ${EXAMPLE_DIR}/namespace-garden.yaml
+kubectl apply -f ${EXAMPLE_DIR}/namespace-garden-core.yaml
+
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: default-domain
+  namespace: garden
+  labels:
+    garden.sapcloud.io/role: default-domain
+  annotations:
+    dns.garden.sapcloud.io/provider: unmanaged
+    dns.garden.sapcloud.io/domain: ${MINIKUBE_IP}.nip.io
+    dns.garden.sapcloud.io/hostedZoneID: ABCDEFG
+type: Opaque
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: internal-domain-nip-io
+  namespace: garden
+  labels:
+    garden.sapcloud.io/role: internal-domain
+  annotations:
+    dns.garden.sapcloud.io/provider: unmanaged
+    dns.garden.sapcloud.io/domain: ${MINIKUBE_IP}.nip.io
+    dns.garden.sapcloud.io/hostedZoneID: ABCDEFG
+type: Opaque
```

Comment:
We shouldn't create these domain secrets from the `setup` script. Instead, I would envision a separate paragraph in the installation instructions mentioning that at least the internal domain secret must be created based on the example in the `example` folder.

We can take the two secrets here and add them (together with your Vagrant PR) additionally to the list of examples so that users can choose which kind of development scenario (only pure locally against Vagrant or also remote against clouds) they want to establish.